PC/winsound.c

```diff
@@ -95,13 +95,15 @@ winsound_PlaySound_impl(PyObject *module, PyObject *sound, int flags)
         }
         wsound = (wchar_t *)view.buf;
     } else {
-        if (!PyUnicode_Check(sound)) {
+        PyObject *obj;
+        if (PyBytes_Check(sound) || (obj = PyOS_FSPath(sound)) == NULL) {
```

Comment:
Thanks for updating the wording! I looked at this again and realized there's another issue:

`PyOS_FSPath` can return a bytes object. I think this won't cause a crash because `PyUnicode_AsWideCharString` will throw a nice TypeError, but we should have tests for passing bytes or a PathLike that resolves to bytes.

Reply:
Thanks. I think this case gives the following error:

```======================================================================
ERROR: test_snd_filepath_as_bytes (__main__.PlaySoundTest.test_snd_filepath_as_bytes)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\Users\msoft\gits\mycpython\Lib\test\test_winsound.py", line 136, in test_snd_filepath_as_bytes
    safe_PlaySound(path, winsound.SND_FILENAME | winsound.SND_NODEFAULT)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\msoft\gits\mycpython\Lib\test\test_winsound.py", line 24, in wrapper
    ret = func(*args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^
TypeError: bad argument type for built-in operation

----------------------------------------------------------------------
Ran 1 test in 0.001s
```

Seemingly because of this line in AsWideCharStr:

```    
if (!PyUnicode_Check(unicode)) {
        PyErr_BadArgument();
        return NULL;
    }
```

shall i add a more helpful error message as a special case?

Reply:
Yes, I'd suggest first calling `PyOS_FSPath`, even if it's bytes, then throwing an error if the result is a bytes.

Reply:
thanks, i've given that a a try